design-system/materials/internals/story/shared-components.js

```diff
@@ -0,0 +1,270 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+/* eslint-disable jsx-a11y/anchor-has-content */
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+import upperFirst from 'lodash/upperFirst';
+import deprecatedTokens from '../deprecated-tokens';
+
+const getIsDeprecated = (token) => deprecatedTokens.includes(token);
+
+const filterGroupItemsValues = (groupItems, searchText) =>
+  Object.entries(groupItems).filter(
+    ([key, value]) =>
+      key.toLowerCase().includes(searchText.toLowerCase()) ||
+      value?.toLowerCase?.().includes(searchText.toLowerCase()) ||
+      value?.description?.toLowerCase().includes(searchText.toLowerCase())
+  );
+
+const Table = styled.table`
+  border: 1px solid #ccc;
+  border-collapse: collapse;
+  & tr td {
+    border: 1px solid #ccc;
+    padding: 15px;
+    text-align: left;
+  }
+  & thead td {
+    background-color: gray;
+    color: white;
+    font-weight: bold;
+  }
+`;
+
+const GroupStyle = styled.div`
+  padding: 10px;
+  display: ${(props) => (props.isVisible ? 'block' : 'none')};
+`;
+
+export const Token = styled.p`
+  font-family: monospace;
+`;
+
+const DeprecationBadge = () => <b style={{ color: 'orange' }}>DEPRECATED</b>;
+DeprecationBadge.displayName = 'DeprecationBadge';
+
+export const TokenBodyCell = (props) => (
+  <>
+    <Token>{props.tokenName}</Token>
+    {getIsDeprecated(props.tokenName) && <DeprecationBadge />}
+  </>
+);
+TokenBodyCell.propTypes = {
+  tokenName: PropTypes.string.isRequired,
+};
+
+export const TokenGroupLinks = (props) => {
```

Comment:
This is a shared component to render the links we have in the header of the story for every design tokens group.